src/frontends/pytorch/src/transforms/string_equality_replacer.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "string_equality_replacer.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/not_equal.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "pt_framework_node.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+using namespace ov::pass;
+using namespace ov::op;
+
+std::string get_str_val(const std::shared_ptr<PtFrameworkNode> fw_node) {
+    FRONT_END_OP_CONVERSION_CHECK(fw_node, "Framework node does not exist.");
+    auto attrs = fw_node->get_attrs();
+    FRONT_END_OP_CONVERSION_CHECK(attrs.find("string_value") != attrs.end(), "Could not retrieve string value.");
+    return attrs.at("string_value");
+}
+
+StringEqualityReplacer::StringEqualityReplacer() {
+    auto framework_node_lhs = pattern::wrap_type<PtFrameworkNode>();
+    auto framework_node_rhs = pattern::wrap_type<PtFrameworkNode>();
+    auto convert_like = pattern::wrap_type<v1::ConvertLike>({framework_node_rhs, framework_node_lhs});
+    auto equal_op = pattern::wrap_type<v1::Equal>({framework_node_lhs, convert_like});
+    auto not_equal_op = pattern::wrap_type<v1::NotEqual>({framework_node_lhs, convert_like});
+
+    auto string_equality_pattern = std::make_shared<pattern::op::Or>(OutputVector{equal_op, not_equal_op});
+
+    ov::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto& pattern_map = m.get_pattern_value_map();
+
+        auto lhs_node =
+            std::dynamic_pointer_cast<PtFrameworkNode>(pattern_map.at(framework_node_lhs).get_node_shared_ptr());
+        if (!lhs_node) {
+            return false;
+        }
+        std::string lhs = get_str_val(lhs_node);
+
+        auto rhs_node =
+            std::dynamic_pointer_cast<PtFrameworkNode>(pattern_map.at(framework_node_rhs).get_node_shared_ptr());
+        if (!rhs_node) {
+            return false;
+        }
+        std::string rhs = get_str_val(rhs_node);
```

Comment:
If input to `Equal` or `NotEqual` is not a string, but some other `PtFrameworkNode` operation this will throw. We should avoid throwing in transformations, better to return `false`.

Reply:
Done